src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/RemoteVettingController.php

```diff
@@ -93,14 +100,12 @@ public function remoteVetAction(Request $request, $secondFactorId, $identityProv
         }
 
         $secondFactor = $service->findOneVerified($secondFactorId);
-        if ($secondFactor === null) {
+        if ($secondFactor === null || $this->expirationHelper->hasExpired($secondFactor->registrationRequestedAt)) {
```

Comment:
Good call to use the expirationHelper here! Would it not be a good defense in depth measure to also verify the token has not expired in the `remoteVetMatchAction`? Highly unlikely the token expired in the meantime, but it is certainly possible.

Reply:
I deliberately didn't add a check on the match page to be user friendly. If a RV process is started in time then why not let it finish. Then it would be better to let a user not even start in the first place. The matching itself could now possibly take long but that shouldn't be happening when the final algorithm is implemented.

I certainly agree we have to make a informed decision about this. I'll add your concerns to the things to do after MVP so we keep this in scope and could let our PM decide what should be guarded.

Be aware that the time of expiration doesn't seem to be so clear either:
https://www.pivotaltracker.com/story/show/90087146

Reply:
I'm fine with implementing it at a later stage, but I do think that when you somehow drop into the second portion of the verification, you should also have this token expired check. To be clear, in __addition__ to the check you already implemented!

But as long as we have this covered on the backlog im all :+1: